e2e-tests/smart-update3/run

```diff
@@ -0,0 +1,214 @@
+
+	# Update cluster to use version service with short schedule
+	kubectl_bin patch pxc/"${CLUSTER}" --type=merge -p '{"spec":{"upgradeOptions":{"versionServiceEndpoint":"'${VS_ENDPOINT}'","apply":"recommended","schedule": "* * * * *"}}}'
+
+	# Wait for PMM2 update
+	wait_pmm_update "${CLUSTER}" "${CLUSTER_SIZE}" "${pmm2_recommended_image}" "2" || exit 1
+
+	wait_cluster_consistency "${CLUSTER}" "${CLUSTER_SIZE}" "${PROXY_SIZE}"
+
+	# Verify PMM2 updated
+	for i in $(seq 0 $((CLUSTER_SIZE - 1))); do
+		actual_pmm_image=$(kubectl_bin get pod "${CLUSTER}-pxc-${i}" -o jsonpath='{.status.containerStatuses[?(@.name=="pmm-client")].image}')
+		if [[ "${actual_pmm_image}" != *"${pmm2_recommended_image}"* ]]; then
+			echo "ERROR: PMM2 image not updated on ${CLUSTER}-pxc-${i}. Expected: ${pmm2_recommended_image}, Got: ${actual_pmm_image}"
+			exit 1
+		fi
+	done
+	desc "PMM2 successfully updated to ${pmm2_recommended_image}"
+
+	kubectl_bin delete -f "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+	kubectl_bin delete pvc --all
+
+	##################################################
+	desc 'PMM3 cluster update with the recommended image by version service'
+
+	desc "Updating secret for PMM3 (pmmserver -> pmmservertoken)"
+	# Get current pmmserver value from my-cluster-secrets
+	pmm_password=$(kubectl_bin get secret my-cluster-secrets -o jsonpath='{.data.pmmserver}')
+
+	# Patch my-cluster-secrets: remove pmmserver, add pmmservertoken
+	kubectl_bin patch secret my-cluster-secrets --type=json \
+		-p '[{"op":"remove","path":"/data/pmmserver"},{"op":"add","path":"/data/pmmservertoken","value":"'${pmm_password}'"}]'
+
+	# Prepare PMM3 cluster config
+	cp -f "${test_dir}/conf/${CLUSTER}-pmm3.yml" "${tmp_dir}/${CLUSTER}-pmm3.yml"
+	yq -i eval ".spec.initContainer.image = \"${IMAGE}\"" "${tmp_dir}/${CLUSTER}-pmm3.yml"
+	spinup_pxc "${CLUSTER}" "${tmp_dir}/${CLUSTER}-pmm3.yml"
+
+	# Get actual PXC version from running cluster and add to version service
+	wait_cluster_consistency "${CLUSTER}" "${CLUSTER_SIZE}" "${PROXY_SIZE}"
+	ACTUAL_PXC_VERSION=$(kubectl_bin get pxc "${CLUSTER}" -o jsonpath='{.status.pxc.version}')
+	desc "Detected PXC version for PMM3: ${ACTUAL_PXC_VERSION}"
+	add_pxc_version_to_vs "${ACTUAL_PXC_VERSION}" "${IMAGE_PXC}"
+
+	# Get initial PMM3 image
+	initial_pmm3_image=$(kubectl_bin get pod "${CLUSTER}-pxc-0" -o jsonpath='{.status.containerStatuses[?(@.name=="pmm-client")].image}')
+	desc "Initial PMM3 image: ${initial_pmm3_image}"
+
+	# Get recommended PMM3 image from version service
+	pmm3_recommended_image=$(kubectl_bin exec -ti "$(get_operator_pod)" ${OPERATOR_NS:+-n $OPERATOR_NS} -- curl -s "${VS_URL}.${namespace}.svc.cluster.local:${VS_PORT}/versions/v1/pxc-operator/9.9.9" | jq -r '.versions[].matrix.pmm | to_entries[] | select(.value.imagePath) | select(.value.imagePath | contains("pmm-client:3")) | select(.value.status == "recommended") | .value.imagePath')
+	desc "Target PMM3 recommended image: ${pmm3_recommended_image}"
```

Comment:
Maybe it makes sense to add a check that the initial does not equal to recommended? Otherwise the update will not happen and the test will not check anything.